enterprise/api/v1/views.py

```diff
@@ -82,6 +84,26 @@ class EnterpriseCustomerUserViewSet(EnterpriseReadOnlyModelViewSet):
     filter_fields = FIELDS
     ordering_fields = FIELDS
 
+    @detail_route()
+    def entitlements(self, request, pk=None):  # pylint: disable=invalid-name,unused-argument
+        """
+        Retrieve the list of entitlements available to this learner.
+
+        Only those entitlements are returned that satisfy enterprise customer's data sharing setting.
+
+        Arguments:
+            request (HttpRequest): Reference to in progress request instance.
+            pk (Int): value if the primary key for the selected enterprise learner
+
+        Returns:
+            (HttpResponse): Response object containing a list of learner's entitlements.
+        """
+        enterprise_customer_user = self.get_object()
+
+        instance = {"entitlements": enterprise_customer_user.entitlements}
```

Comment:
`self.get_object` returns an object or [raises a `404` error](https://github.com/tomchristie/django-rest-framework/blob/master/rest_framework/generics.py#L97), so we are guaranteed to have an object at line 103. And documentation of DRF also follows this pattern. (Second example on this link http://www.django-rest-framework.org/api-guide/routers/#example)
```
    @detail_route()
    def group_names(self, request):
        """
        Returns a list of all the group names that the given
        user belongs to.
        """
        user = self.get_object()
        groups = user.groups.all()
        return Response([group.name for group in groups])
```
So, I think it would be redundant to use getattr. let me know if you had something else in mind